### PR TITLE
py-smote-variants: Added package py-smote-variants

### PR DESCRIPTION
**Pull request by alex391**
Also added py-minisom and py-metric-learn as dependencies

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-metric-learn/package.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    scikit-learn, the leading library for machine learning in Python. This
+    allows to use all the scikit-learn routines (for pipelining, model
+    selection, etc) with metric learning algorithms through a unified
+    interface."""
+
+    homepage = "https://github.com/scikit-learn-contrib/metric-learn"
+    pypi = "metric-learn/metric-learn-0.7.0.tar.gz"
+
+    version("0.7.0", sha256="2b35246a1098d74163b16cc7779e0abfcbf9036050f4caa258e4fee55eb299cc")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-metric-learn/package.py:22]: Missing build dep on setuptools

File: var/spack/repos/builtin/packages/py-smote-variants/package.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    depends_on("py-wheel@0.33.4:", type=("build"))
+
+    depends_on("py-setuptools@41.0.1:", type=("build"))
+
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"))
+    depends_on("py-scikit-learn", type=("build", "run"))
+    depends_on("py-joblib", type=("build", "run"))
+    depends_on("py-minisom", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-smote-variants/package.py:28]: Missing statistics dep

File: var/spack/repos/builtin/packages/py-smote-variants/package.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    """Variants of the synthetic minority oversampling technique (SMOTE) for
+    imbalanced learning"""
+
+    homepage = "https://github.com/analyticalmindsltd/smote_variants"
+    pypi = "smote_variants/smote_variants-0.7.3.tar.gz"
+
+    version("0.7.3", sha256="69497c764f101a76e8a3d4a9c80176704375c7aa5e26914f19222b59fb03b890")
+
+    depends_on("py-wheel@0.33.4:", type=("build"))
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-smote-variants/package.py:19]: Missing pytest-runner build dep

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by alex391** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-smote-variants/package.py (hunk 2 above)
Comment (alex391) [reply var/spack/repos/builtin/packages/py-smote-variants/package.py:28]: Statistics is only required for Python 2 (it's built-in to 3) — I've added a comment to make that clear. The package also needs `python@3.5:` anyway.
Thank you for the feedback!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**